src/components/LeaveSessionPrompt/LeaveSessionPrompt.tsx

```diff
@@ -13,13 +13,13 @@ const blockUserLeavingPage = ({
   currentLocation,
   nextLocation,
 }: ShouldBlockParams) => {
-  // allowing user to view subjects pages during reviews and to review summary page
-  const subjDetailsRegex = new RegExp("/subjects/*");
   if (
-    subjDetailsRegex.test(nextLocation.pathname) ||
-    nextLocation.pathname === "/reviews/summary" ||
-    nextLocation.pathname === "/lessons/quiz" ||
-    nextLocation.pathname === "/lessons/summary"
+    // allowing user to view subjects pages during reviews and to review summary page
+    nextLocation.pathname.startsWith("/subjects") ||
+    // Don't block redirection to / from summaries
+    nextLocation.pathname.endsWith("/summary") ||
+    currentLocation.pathname.endsWith("/summary") ||
+    nextLocation.pathname.endsWith("/quiz")
```

Comment:
It looks like this has an issue where it doesn't account for navigating back from subject details page to a review or lesson session, see video:

https://github.com/salemlf/hakubun/assets/25971591/9e7c7cac-8c47-48ca-955c-da0322a075a3

However, seems like an easy fix! If you add an additional condition to check for navigating to a session: `|| nextLocation.pathname.endsWith("/session")`, this fixes the bug. After adding that condition: https://github.com/salemlf/hakubun/assets/25971591/6df37971-e178-44c0-8026-ea1b19737a6f

Reply:
Nice catch! I didn't run into that when I checked it but I'll add the fix now!